python/tk_desktop/project_commands_model.py

```diff
@@ -306,52 +286,29 @@ def _handle_command_triggered(self, item, command_name=None, button_name=None,
+                'launch_nuke': {
+                    'timestamp': datetime.datetime(2016, 5, 20, 21, 48, 17, 495234),
+                    'added': False},
+                ...
+            }
         """
-        # dictionary to keep track of the commands launched with recency information
-        # each command name keeps track of a timestamp of when it was last launched
-        # and a boolean saying whether the corresponding command has been registered
-        self.__recents = {}
+        recents = {}
+        for name, details in self.__recents.iteritems():
+            recents[name] = {"timestamp": details["timestamp"], "added": False}
+            self._settings_manager.store("sg_desktop.recent_apps", recents,
+                                         self._settings_manager.SCOPE_PROJECT)
 
-        # pull down matching invents for the current project for the current user
-        filters = [
-            ["user", "is", sgtk.platform.current_engine().get_current_login()],
-            ["project", "is", self.__project],
-            ["event_type", "is", self.APP_LAUNCH_EVENT_TYPE],
-        ]
-
-        # execute the Shotgun summarize command
-        # get one group per description with a summary of the latest created_at
-        engine = sgtk.platform.current_engine()
-        connection = engine.shotgun
-        start_time = time.time()
-        summary = connection.summarize(
-            entity_type="EventLogEntry",
-            filters=filters,
-            summary_fields=[{"field": "created_at", "type": "latest"}],
-            grouping=[{"field": "description", "type": "exact", "direction": "desc"}],
-        )
-        end_time = time.time()
-        call_duration = end_time-start_time
-        engine.log_debug("App Launches summarized (%.3f s)" % call_duration)
-
-        # parse the results
-        for group in summary["groups"]:
-            # convert the text representation of created_at to a datetime
-            text_stamp = group["summaries"]["created_at"]
-            time_stamp = datetime.datetime.strptime(text_stamp, "%Y-%m-%d %H:%M:%S %Z")
-
-            # pull the command name from the description
-            description = group["group_name"]
-            match = re.search("'(?P<name>.+)'", description)
-            if match is not None:
-                name = match.group("name")
-
-                # if multiple descriptions end up with the same name use the most recent one
-                existing_info = self.__recents.setdefault(
-                    name, {"timestamp": time_stamp, "added": False})
-                if existing_info["timestamp"] < time_stamp:
-                    self.__recents[name]["timestamp"] = time_stamp
+    def __load_recents_from_settings(self):
+        """
+        Loads the dict of recently launched apps from the user settings. See above for the
+        format.
+        """
+        self.__recents = self._settings_manager.retrieve("sg_desktop.recent_apps", {},
```

Comment:
So whenever someone clicks into a project for the first time (including one they just setup), and/or hasn't ever launched anything, it will do an event log lookup?